Fullstack model implementation

Kris2ffer opened the pull request.

Kris2ffer committed:
Added set userlist method

Kris2ffer committed:
Implement pod service

Kris2ffer committed:
Using study service to get studies

Kris2ffer committed:
Using list of study models

Kris2ffer committed:
Update StudyControllerTests.cs

Kris2ffer committed:
Added study info on cards

Kris2ffer committed:
Added pod info on cards

Kris2ffer committed:
Create plus1.svg

Kris2ffer committed:
Update CreateStudyPage.js

Kris2ffer committed:
Updated pod info

Unicus-Tom committed:
Fixed wrong method usage

Unicus-Tom committed:
Made new pod appear first

Kris2ffer committed:
Hide overflow on pod card

Kris2ffer committed:
Merge branch 'Fullstack-model-implementation' of https://github.com/equinor/sepes into Fullstack-model-implementation

Unicus-Tom committed:
Fixed Direction of NSG rules

Unicus-Tom committed:
Hide pods in new study

Unicus-Tom committed:
Added basic rules

Unicus-Tom committed:
Added comment

Kris2ffer committed:
Send study and based

Kris2ffer committed:
Merge branch 'Fullstack-model-implementation' of https://github.com/equinor/sepes into Fullstack-model-implementation

Unicus-Tom committed:
Improved hiding pods

Kris2ffer committed:
Fixed update study

Kris2ffer committed:
Add symbol on new study card

Kris2ffer committed:
Fixed state update

Kris2ffer committed:
Fixed port. String to number.

Kris2ffer committed:
Fixed new study and based study diff

Kris2ffer committed:
Save button no longer hidden

Kris2ffer committed:
Update PodService.cs

Kris2ffer committed:
Set pod id and save button after saving

Kris2ffer committed:
Fixed dataset and archived state

Unicus-Tom committed:
Added alt property to images

Kris2ffer committed:
Fixed error when creating first pod

Kris2ffer committed:
Fixed state when creating new study and pod

Kris2ffer committed:
Update StudyControllerTests.cs

Kris2ffer committed:
Merge branch 'master' into Fullstack-model-implementation

Kris2ffer committed:
Fixed current study state when updating pod

Kris2ffer committed:
Fixed nsg creation when updating pod

gjersvik commented on RestApi/Sepes.RestApi/Services/AzureService.cs:
```diff
@@ -146,6 +150,49 @@ await _azure.NetworkSecurityGroups
+            .WithAnyProtocol()
+            .WithPriority(4000)
+            .Attach()
+            .ApplyAsync();
```
This have be that builder pattern taken to its logical extreme. Should be split up into something smaller.

gjersvik commented on RestApi/Sepes.RestApi/Services/PodService.cs:
```diff
@@ -115,46 +115,37 @@ private async Task ManageNetworkSecurityGroup(Pod newPod, Pod based)
 
             // Delete old nsg
             if (nsgName == nsgNameDefault && nsgNames.Contains(nsgNameDef2)) {
+                Console.WriteLine("#### Deleting old nsg "+nsgNameDef2);
```
Console.WriteLine should not be in master. If debug remove. If not log it using logger.

gjersvik commented on RestApi/Sepes.RestApi/Services/PodService.cs:
```diff
@@ -115,46 +115,37 @@ private async Task ManageNetworkSecurityGroup(Pod newPod, Pod based)
                 await _azure.DeleteSecurityGroup(nsgNameDef2, newPod.resourceGroupName);
             }
             else if (nsgName == nsgNameDef2 && nsgNames.Contains(nsgNameDefault)) {
+                Console.WriteLine("#### Deleting old nsg "+nsgNameDefault);
```
Console.WriteLine should not be in master. If debug remove. If not log it useing logger.

gjersvik commented on RestApi/Sepes.RestApi/Services/StudyService.cs:
```diff
@@ -74,16 +76,19 @@ private Study UpdatePods(Study based, Study study)
                         var newPods = study.pods.Remove(pod).Add(newPod);
                         study = study.ReplacePods(newPods);
 
+                        Console.WriteLine("Create pod with pod id "+newPod.id);
```
Another Console.WriteLine

gjersvik reviewed (changes requested):
Please remove the Console.WriteLine from the code. Also not required but try to spit the mega builder.

Kris2ffer committed:
Removed writelines

Kris2ffer committed:
Update AzureService.cs

Kris2ffer committed:
Fixed issues nsg

Fixed nsg being put in the wrong resource group.
Fixed ports for inbound rules in wrong location.

Kris2ffer committed:
Update PodServiceTests.cs

Unicus-Tom committed:
First draft

First draft of instructions to set up developer environment

Kris2ffer committed:
Added new model-based Studies table

Unicus-Tom committed:
Removed unecesarry files
